sup3r/qa/qa.py

```diff
@@ -272,6 +302,78 @@ def output_handler_class(self):
         elif self.output_type == 'h5':
             return Resource
 
+    def bias_correct_source_data(self, data, feature):
+        """Bias correct data using a method defined by the bias_correct_method
+        input to ForwardPassStrategy
+
+        Parameters
+        ----------
+        data : np.ndarray
+            Any source data to be bias corrected, with the feature channel in
+            the last axis.
+
+        Returns
+        -------
+        data : np.ndarray
+            Data corrected by the bias_correct_method ready for input to the
+            forward pass through the generative model.
+        """
+        method = self.bias_correct_method
+        kwargs = self.bias_correct_kwargs
+        if method is not None:
+            method = getattr(sup3r.bias.bias_transforms, method)
+            logger.info('Running bias correction with: {}'.format(method))
+            feature_kwargs = kwargs[feature]
+            if 'time_index' in signature(method).parameters:
+                feature_kwargs['time_index'] = self.time_index
+            logger.debug('Bias correcting feature "{}" using function: {} '
+                         'with kwargs: {}'
+                         .format(feature, method, feature_kwargs))
+
+            data = method(data, **feature_kwargs)
+
+        return data
+
+    def get_source_dset(self, idf, feature):
+        """Get source low res input data including optional bias correction
+
+        Parameters
+        ----------
+        idf : int
+            Feature index in axis=-1 of the source data handler.
+        feature : str
+            Feature name
+
+        Returns
+        -------
+        data_true : np.array
+            Low-res source input data including optional bias correction
+        """
+
+        height = str(Feature.get_height(feature))
+        uv_comps = np.sum([name.startswith(('U_', 'V_'))
+                           for name in self.source_features
+                           if height in name])
+
+        if 'windspeed' in feature and uv_comps >= 2:
+            u_feat = feature.replace('windspeed_', 'U_')
+            v_feat = feature.replace('windspeed_', 'V_')
+            logger.info('For sup3r output feature "{}", retrieving u/v '
+                        'components "{}" and "{}"'
+                        .format(feature, u_feat, v_feat))
+            u_idf = self.source_handler.features.index(u_feat)
+            v_idf = self.source_handler.features.index(v_feat)
+            u_true = self.source_handler.data[..., u_idf]
+            v_true = self.source_handler.data[..., v_idf]
+            u_true = self.bias_correct_source_data(u_true, u_feat)
+            v_true = self.bias_correct_source_data(v_true, v_feat)
+            data_true = np.hypot(u_true, v_true)
```

Comment:
I always learn about new numpy methods reading your code lol

Reply:
Haha i like this one even though it seems trivial, avoids silly errors like a single * instead of power **